### DISCUSSION TITLE
Improve UX of Role Permissions

### DISCUSSION THREAD

**Discussion by udf2457** [directus/directus#15119] [Feature Requests]
At present the UX of the Role Permissions editor is fairly poor.  Each Collection is independent of each other.

I believe the following capabilities would greatly enhance the UX:

1. "Edit All"

Let's say I'm creating a new Read-Only user, for example for a website frontend.
With the present UX, I have to select "Enable" or "Use Custom" on each individual collection.  If using "Use Custom" this is particularly tiresome.  Being able to say "Enable for all" or "Use this custom rule for all" would greatly enhance the UX.

2. "Copy from Existing"

With the present UX, because each Collection is independent, there is no ability to copy from another.
So, for example, let's say I have a collection `Foo` with some nice custom rules, there is no way to then select collection `Bar` and say "I want to use the rules from `Foo` as a basis" (i.e. bring the rules from `Foo` into the UI, let me review/edit, instead of starting from scratch).

**Comment by benhaynes** [comment]
I completely agree with these two feature additions. Happy to provide mockups when the time is right.

**Comment by azrikahar** [comment]
Linking related discussions here: #11052 #7032 #3008

**Comment by udf2457** [comment]
As a further observation "Field permissions" has to be toggled per field.  There's no "select all" button ?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
